boltstub/scripting.py

```diff
@@ -84,14 +86,22 @@ def auto_match(self, tag):
     def on_auto_match(self, request):
         raise NotImplementedError
 
-    def _get_version(self, request, n):
+    def _get_version_range_defintion(self, request, n):
         versions_offset = 4  # Skip past magic header
         version_size = 4  # Number of bytes reserved per version
+        range_offset = 1  # Minor version offset relative to start index of the version range
         minor_offset = 2  # Minor version offset relative to start of version
         major_offset = 3  # Major version offset relative to start of version
         versionindex = versions_offset + (n * version_size)
         return (request[versionindex + major_offset],
-                request[versionindex + minor_offset])
+                request[versionindex + minor_offset],
+                request[versionindex + range_offset])
+
+    def _get_versions(self, request): 
+        for n in range(0, 4):
+            (major, max_minor, minor_offset) = self._get_version_range_defintion(request, n)
```

Comment:
Can't remember what the spec said... is minor_offset number of prior versions to support or the absolute min version? Judging by max_minor - minor_offset I suppose it is the number of back versions. Is it really right with an extra -1 in range?

Reply:
minor_offset is the number of previous minor versions ([see](https://docs.google.com/document/d/1BQraZtw0_o_tEezCJE8-5RsL6d03Lg3bXTezwwIdUy4/edit)  

The range is open to the right, so a range(1, 2) will be [1], (1, 3) will be [1, 2], (3, 2, -1) will be [3]

Reply:
Looks fine! Maybe a better name for minor_offset, it makes sense when offsetting in request but not so much here.
I like the look of this in on_handshake!

Reply:
I've change the name to minor_difference, since it's the name used in the spec.